src/test/regress/expected/failure_1pc_copy_hash.out

```diff
@@ -0,0 +1,374 @@
+SELECT citus.mitmproxy('flow.allow()');
+ mitmproxy 
+-----------
+ 
+(1 row)
+
+SET citus.shard_count = 1;
+SET citus.shard_replication_factor = 2; -- one shard per worker
+SET citus.multi_shard_commit_protocol TO '1pc';
+SET citus.next_shard_id TO 100400;
+ALTER SEQUENCE pg_catalog.pg_dist_placement_placementid_seq RESTART 100;
+CREATE TABLE copy_test (key int, value int);
+SELECT create_distributed_table('copy_test', 'key');
+ create_distributed_table 
+--------------------------
+ 
+(1 row)
+
+COPY copy_test FROM PROGRAM 'echo 0, 0 && echo 1, 1 && echo 2, 4 && echo 3, 9' WITH CSV;
+SELECT count(1) FROM copy_test;
+ count 
+-------
+     4
+(1 row)
+
+-- ==== kill the connection when we try to start a transaction ====
+-- the query should abort
+SELECT citus.mitmproxy('flow.contains(b"assign_distributed_transaction").killall()');
+ mitmproxy 
+-----------
+ 
+(1 row)
+
+COPY copy_test FROM PROGRAM 'echo 0, 0 && echo 1, 1 && echo 2, 4 && echo 3, 9' WITH CSV;
+WARNING:  connection not open
+CONTEXT:  while executing command on localhost:57640
+COPY copy_test, line 1: "0, 0"
+ERROR:  failure on connection marked as essential: localhost:57640
+CONTEXT:  COPY copy_test, line 1: "0, 0"
+-- ==== kill the connection when we try to start the COPY ====
+-- the query should abort
+SELECT citus.mitmproxy('flow.contains(b"FROM STDIN WITH").killall()');
```

Comment:
> Do we always need to execute SELECT citus.mitmproxy('flow.allow()'); before any new killall or is it safe as we do here? '

We need a `flow.allow()` after any call to `flow.[things].killall()`, because once `killall` triggers it prevents any more connections from being made. If the last call was `kill` then we're safe, because only connections which include the thing `kill` matches against will be killed.

> Another question: Do we have anything like flow.disallow() or such to test the case where the coordinator cannot reach to the worker at all? Is that killall ?

Yep!

That's `flow.kill()` or `flow.killall()`, both do the same thing.

Reply:
`flow.reset()` probably does a better job of simulating a worker which can't be reached at all, but I don't think it makes a difference for us, `libpq` makes all the failure modes look the same.